packages/slice-machine/src/features/aiFeedback/index.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  AnimatedElement,
+  Box,
+  IconButton,
+  Portal,
+  Text,
+  Toast,
+  Tooltip,
+} from "@prismicio/editor-ui";
+import { useState } from "react";
+import { z } from "zod";
+
+import { telemetry } from "@/apiClient";
+import { usePersistedState } from "@/hooks/usePersistedState";
+
+export function addAiFeedback({
+  type,
+  library,
+  sliceId,
+  variationId,
+  langSmithUrl,
+}: {
+  type: "model";
+  library: string;
+  sliceId: string;
+  variationId: string;
+  langSmithUrl?: string;
+}) {
+  const key = getKey({ type, library, sliceId, variationId });
+  const feedback = JSON.stringify({ langSmithUrl });
+  localStorage.setItem(key, feedback);
+}
+
+export function useAiFeedback({
+  type,
+  library,
+  sliceId,
+  variationId,
+}: {
+  type: "model";
+  library: string;
+  sliceId: string;
+  variationId: string;
+}) {
+  const key = getKey({ type, library, sliceId, variationId });
+  const [value, setValue] = usePersistedState(key, undefined, {
+    schema: z.object({
+      langSmithUrl: z.string().url().optional(),
+    }),
+  });
+  return {
+    key,
+    value,
+    remove: () => setValue(undefined),
```

Comment:
❓ This will keep the key in the local storage, no? Just without a langSmithUrl.
I find it a bit strange that we rely on the value to decide to display or not, I would use the key for that. And ensure to clean the key after. langSmith is a detail, but is used as the source of truth.

Reply:
When the value is undefined, the whole item is removed from local storage. I changed the implementation of the hook in the previous PR to account for this.

Reply:
Dam I should have check the other PR, perfect then!